include/reporters/catch_reporter_bases.hpp

```diff
@@ -18,13 +18,30 @@
 #include <cassert>
 #include <memory>
 #include <ostream>
+#include <sstream>
 
 namespace Catch {
     void prepareExpandedExpression(AssertionResult& result);
 
     // Returns double formatted as %.3f (format expected on output)
     std::string getFormattedDuration( double duration );
 
+    template<typename Container>
+    std::string containerToString( Container const& container, const char seperator = ' ' ) {
```

Comment:
> I added some review notes.
> 
> Also please don't merge master -> feature branch, rebase the feature branch instead.

I can't change the base branch in this PR to use my master (which is not rebased with this feature). Is there a way to do that or should I open a new PR again?

Reply:
Also, please ignore the force-push (I tried to set my master branch's upstream to my feature branch and push the changes so that this PR will actually be a rebase to catchorg/master).